presqt/osf/classes/file.py

```diff
@@ -14,20 +14,34 @@ def _populate_attributes(self, file):
         file : dict
             Data dictionary returned from the json response to create the File class instance.
         """
-        if not file:
-            return
-
         self.id = file['id']
+        # Links
         self._endpoint = file['links']['self']
         self._download_url = file['links']['download']
         self._upload_url = file['links']['upload']
         self._delete_url = file['links']['delete']
-        self.osf_path = file['attributes']['path']
-        self.path = file['attributes']['materialized_path']
-        self.name = file['attributes']['name']
-        self.date_created = file['attributes']['date_created']
-        self.date_modified = file['attributes']['date_modified']
-        self.hashes = file['attributes']['extra']['hashes']
+        # Attributes
```

Comment:
These attributes should *probably* be declared in an __init__() method and then populated with this method.

Generally, we don't want to create new object atrributes without declaring them ahead of time... not because the language requires it, but because it gives a consistent place for other developers to understand what attributes a given class has.

Reply:
Oh second thought... perhpas _populate_attributes can just go away and this action would just take place in the __init__() method of your child classes like this one.

We can chat if this doesn't make any sense.